### PR TITLE
net: openthread: Add config for NCP vendor hook

### PR DESCRIPTION
**Pull request by Philipp-Wohlgenannt-Tridonic-com**
Add config for NCP vendor hook.

### CODE REVIEW

File: subsys/net/l2/openthread/Kconfig [hunk 1]
```diff
@@ -261,6 +261,18 @@ config OPENTHREAD_NCP_RADIO
...
 	  Enable NCP in OpenThread stack as radio-only
 
+config OPENTHREAD_NCP_VENDOR_HOOK
+	bool "Enable NCP vendor hook"
+        depends on OPENTHREAD_NCP
+        help
+	  Enable vendor specific code for NCP
+
+config OPENTHREAD_NCP_VENDOR_HOOK_SOURCE
+	string "NCP vendor hook source file"
```
Comment (rlubos) [inline subsys/net/l2/openthread/Kconfig:271]: I think this entry should be depending on OPENTHREAD_NCP_VENDOR_HOOK, i. e. 
`depends on OPENTHREAD_NCP_VENDOR_HOOK`

**Review by rlubos — COMMENTED** [review]
Please update OpenThread revision in the `west.yml` with a PR reference from OpenThread repository (`pull/13/head`). Also note that due to feature freeze in Zephyr, this PR may take a while.

**Comment by rlubos** [comment]
Also please fix the checkpatch warnings.

File: subsys/net/l2/openthread/Kconfig (hunk 1 above)
Comment (Philipp-Wohlgenannt-Tridonic-com) [reply subsys/net/l2/openthread/Kconfig:265]: I have reduced it to only one option

**Comment by Philipp-Wohlgenannt-Tridonic-com** [comment]
-:11: WARNING:LEADING_SPACE: please, no spaces at the start of a line
#11: FILE: subsys/net/l2/openthread/Kconfig:269:
+        depends on OPENTHREAD_NCP$

-:12: WARNING:LEADING_SPACE: please, no spaces at the start of a line
#12: FILE: subsys/net/l2/openthread/Kconfig:270:
+        help$

- total: 0 errors, 2 warnings, 18 lines checked

I don't know how to fix this warnings. I don't see any difference to the existing code style. @rlubos what should i do with these 2 warnings?

**Comment by rlubos** [comment]
@Philipp-Wohlgenannt-Tridonic-com 

> All checks are passing now.

It seems to be fine now. The warnings you mention apparently refer to some earlier revision.

File: subsys/net/l2/openthread/Kconfig (hunk 1 above)
Comment (rlubos) [reply subsys/net/l2/openthread/Kconfig:265]: Ok, so now this config will always be enabled when we enable NCP. Will it work, if we don't want to use vendor hook and we leave this option as an emtpy string?

Comment (Philipp-Wohlgenannt-Tridonic-com) [reply subsys/net/l2/openthread/Kconfig:265]: Yes it works with an empty string.
With an empty string the if from https://github.com/zephyrproject-rtos/openthread/pull/13/commits/35d3bdfa71f850de764dbdb3ec01252d4cb853c8 doesn't set the option OT_NCP_VENDOR_HOOK_SOURCE so openthread is not using or expecting a ncp vendor hook.

Comment (rlubos) [reply subsys/net/l2/openthread/Kconfig:265]: OK, I see, thanks.

**Review by rlubos — APPROVED** [review]

**Comment by rlubos** [comment]
DNM until OpenThread is updated

**Review by tbursztyka — APPROVED** [review]

**Comment by rlubos** [comment]
The OpenThread update and https://github.com/zephyrproject-rtos/openthread/pull/13 were merged, please update SHA: 981b502fc674121fe70ee119de8384acae050f6d

**Comment by kkasperczyk-no** [comment]
@Philipp-Wohlgenannt-Tridonic-com are you planning to make any further activities connected with this pull request and inserting changes stated by @rlubos? I'm interested in inserting the same changes, as you did in your commit.

**Comment by markus-becker-tridonic-com** [comment]
> @Philipp-Wohlgenannt-Tridonic-com are you planning to make any further activities connected with this pull request and inserting changes stated by @rlubos? I'm interested in inserting the same changes, as you did in your commit.

@Philipp-Wohlgenannt-Tridonic-com will be back in office towards end of the month.

**Comment by rlubos** [comment]
@Philipp-Wohlgenannt-Tridonic-com The `west.yml` is already up to date, and `OPENTHREAD_NCP_VENDOR_HOOK_SOURCE` was added in some other PR, I think we can close this one.

**Comment by Philipp-Wohlgenannt-Tridonic-com** [comment]
> @Philipp-Wohlgenannt-Tridonic-com The `west.yml` is already up to date, and `OPENTHREAD_NCP_VENDOR_HOOK_SOURCE` was added in some other PR, I think we can close this one.

Yes we can close this request. Can you do that?

**Comment by rlubos** [comment]
Sure, thanks!

**Pull request closed without merging**